File: RobloxHybrid/platforms/android/Hybrid/hybridlib/src/main/java/com/roblox/hybrid/modules/RBHybridModuleAnalytics.java
package com.roblox.hybrid.modules;

import com.roblox.hybrid.RBHybridCommand;
import com.roblox.hybrid.RBHybridModule;

/**
 * Created by roblox on 3/6/15.
 */
public class RBHybridModuleAnalytics extends RBHybridModule {
    private static final String MODULE_ID = "Analytics";

    public RBHybridModuleAnalytics() {
        super(MODULE_ID);
    }

    @Override
    public void execute(RBHybridCommand command) {
        command.executeCallback(true, null);
    }
}

File: RobloxHybrid/platforms/ios/Modules/analytics/RBHybridModuleAnalytics.m
//
//  RBHybridModuleDialogs.m
//  RobloxMobile
//
//  Created by Ariel Lichtin on 2/5/15.
//  Copyright (c) 2015 ROBLOX. All rights reserved.
//

#import "RBHybridModuleAnalytics.h"
#import "RBHybridCommand.h"
#import "NSDictionary+Parsing.h"
#import "Flurry.h"

#define MODULE_ID @"Analytics"

@implementation RBHybridModuleAnalytics

-(instancetype) init
{
    self = [super initWithModuleID:MODULE_ID];
    if(self)
    {
        [self registerFunction:@"trackEvent" withSelector:@selector(trackEvent:)];
    }
    return self;
}

/**
 * Present a single button dialog
 *
 * @param {string} eventName - Event name
 * @param {Object} params - Dictionary with the event parameters. Only the first level will be saved. (i.e. no nested objects).
 *
 */
-(void)trackEvent:(RBHybridCommand*)command
{
    NSString* event = [command.params stringForKey:@"eventName" withDefault:@"Roblox"];
    NSDictionary* params = [command.params dictionaryForKey:@"params" withDefault:@{}];
    
    [Flurry logEvent:event withParameters:params];
}

@end
